.github/workflows/pr.yaml

```diff
@@ -3,145 +3,48 @@ name: Review
 jobs:
   test-composer-install:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        php: [ '8.4' ]
-    name: Validate composer (${{ matrix.php }})
+    name: Validate composer
     steps:
       - uses: actions/checkout@v4
-
-      - name: Setup PHP, with composer and extensions
-        uses: shivammathur/setup-php@v2
-        with:
-          php-version: ${{ matrix.php }}
-          extensions: http, ctype, iconv
-          coverage: none
-
-      - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
-
-      - name: Cache dependencies
-        uses: actions/cache@v4
-        with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ matrix.dependency-version }}-
-          restore-keys: ${{ runner.os }}-composer-${{ matrix.dependency-version }}-
-
-      - name: Composer install with exported .env variables
-        run: |
-          set -a && source .env && set +a
-          APP_ENV=prod composer install --no-dev -o
-
-      - name: Validate composer files
-        run: |
-          composer validate composer.json --strict
+      - run: |
           docker network create frontend
+          docker compose run --user root --rm phpfpm composer validate composer.json --strict
```

Comment:
Consider adding

``` yaml
env:
  COMPOSE_USER: runner
```

to get rid of `--user root` in all `docker compose run` (or `… exec`) incantations.